### ISSUE TITLE
GH Issues as writing idea management tool

### ISSUE THREAD

**Issue by pglevy** [pglevy/writers-repo#3]
Use GitHub Issues as a quick way to capture ideas and build on them.

GitHub Issues are one of the core features of GitHub for tracking bugs, suggesting enhancements, and managing work for software projects. 

### Background

I was talking to my wife the other day about her starting a blog. My proposal was to keep it simple and use a basic website hosted on GitHub Pages. Even though it wouldn't take long to set up, I wanted to encourage her to start capturing ideas for content now. I didn't want her to be blocked by getting a site up and running. And although she had already been jotting down thoughts in a notebook, I thought that getting her ideas onto the screen would help make it feel more real.

So after thinking about it for a few minutes, I suggested she simply create an Issue in the GitHub repo we set up for her blog. This is why I like this idea as a starting point:

- The content editor includes some basic formatting tools.
- You capture the content in Markdown, so it's reusable, but you can also preview it immediately.
- You can edit the content and it tracks the version history.
- You can add comments, so we could also use it for feedback while refining ideas.
- You can drag and drop images for automatic uploading and hosting.

After playing around with this idea, I realized there are even more powerful benefits to using Issues as a writing idea management tool:

- Cross-reference ideas by typing `#`, which you gives you a searchable auto-complete of all your other issues.
- Use Labels to group ideas around different purposes or themes.
- Use Projects to track a content workflow, for example, Writing, Editing, Proofreading, Published
- Use the GitHub mobile app to capture and edit content. (You can't even do that with Markdown files in the repo itself.)
- Create writing prompts using Issue Templates and Forms. 🤯
- Add contributors to your project and collaborate by assigning issues to each other at different point in the process.

### Other cool stuff

- Issues aren't in the repo, but you can export so your content is never trapped in one platform.
- Issue templates are in the repo version control, so it's possible to collaborate on them and share.
- You can do this with a public repo and build in public or capture your thoughts in a private repo. Or both!
- Every idea gets its own URL, a permanent link you can use to reference it from anywhere (unless you delete the issue).

### Downsides

- Because you're working the browser, it does require a live internet connection. (This goes for the mobile app too.)
- While you do get version tracking on the original issue content, it doesn't allow you add any "change notes." You just get a list of versions sorted by recency.

### The Writer's Repo

So now I've really started geeking on this issues-as-writing-ideas approach and plan to use it moving forward. If you're interested in trying this out, I've set up a Writer's Repo template in the GitHub for No-Coders org you can use as a starting point. It contains the following templates to help structure some specific writing needs:
- Tweet: includes a 280-character comment so you can visually see about how long it should be.
- Blog: set up as an input form with high-level prompts to structure your thoughts.
- Pitch: a form based on the Basecamp approach to articulating product ideas.
- Newsletter: a fairly open template with some prompts about a call-to-action and resource links.

Here's all you need to do to get writing:
1. Copy the repo using the `Use this template` button.
2. Go to the Issues tab and press the `New issue` button.
3. Select one of the templates or start with a blank issue.

Features:
- editing (version control)
- commenting
- labeling
- project management, with automation
- mobile app (for capturing formatted content)
  - can even add images!
- mobile sharing

![20220112_142606.jpg](https://user-images.githubusercontent.com/5177337/149336736-20d9e0e2-6f15-4001-b524-4435f12cae11.jpg)
